src/dsl/cb.rs

```diff
@@ -148,13 +209,24 @@ pub fn select<
     let when_true = when_true.into();
     let when_false = when_false.into();
 
+    match selector.typing {
+        Typing::Boolean => (),
+        _ => panic!("Expected boolean selector, got {:?}, selector: {}", selector.typing, selector.annotation)
+    }
+
+    match (when_true.typing, when_false.typing) {
```

Comment:
when_true and when_false can be anything. If they are both the same type that is the returned type, otherwise unknown

Reply:
Fixed